pkg/compute/build.go

```diff
@@ -129,17 +124,168 @@ func (c *BuildCommand) Exec(in io.Reader, out io.Writer) (err error) {
 	progress.Step("Creating package archive...")
 
 	dest := filepath.Join("pkg", fmt.Sprintf("%s.tar.gz", name))
-	err = createPackageArchive([]string{
+
+	files := []string{
 		ManifestFilename,
 		"Cargo.toml",
-		"bin",
-		"src",
-	}, dest)
+	}
+
+	ignoreFiles, err := getIgnoredFiles(IgnoreFilePath)
+	if err != nil {
+		return err
+	}
+
+	binFiles, err := getNonIgnoredFiles("bin", ignoreFiles)
 	if err != nil {
 		return err
 	}
+	files = append(files, binFiles...)
+
+	if c.includeSrc {
+		srcFiles, err := getNonIgnoredFiles("src", ignoreFiles)
+		if err != nil {
+			return err
+		}
+		files = append(files, srcFiles...)
+	}
+
+	err = createPackageArchive(files, dest)
+	if err != nil {
+		return fmt.Errorf("error creating package archive: %w", err)
+	}
 
 	progress.Done()
+
 	text.Success(out, "Built %s package %s (%s)", lang, name, dest)
 	return nil
 }
+
+// createPackageArchive packages build artifacts as a Fastly package, which
+// must be a GZipped Tar archive such as: package-name.tar.gz.
+//
+// Due to a behavior of archiver.Archive() which recursively writes all files in
+// a provided directory to the archive we first copy our input files to a
+// temporary directory to ensure only the specified files are included and not
+// any in the directory which may be ignored.
+func createPackageArchive(files []string, destination string) error {
+	// Create temporary directory to copy files into.
+	p := make([]byte, 8)
+	n, err := rand.Read(p)
+	if err != nil {
+		return fmt.Errorf("error creating temporary directory: %w", err)
+	}
+
+	tmpDir := filepath.Join(
+		os.TempDir(),
+		fmt.Sprintf("fastly-build-%x", p[:n]),
+	)
+
+	if err := os.MkdirAll(tmpDir, 0700); err != nil {
+		return fmt.Errorf("error creating temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// Create implicit top-level directory within temp which will become the
+	// root of the archive. This replaces the `tar.ImplicitTopLevelFolder`
+	// behavior.
+	dir := filepath.Join(tmpDir, fileNameWithoutExtension(destination))
+	if err := os.Mkdir(dir, 0700); err != nil {
+		return fmt.Errorf("error creating temporary directory: %w", err)
+	}
+
+	for _, src := range files {
+		dst := filepath.Join(dir, src)
+		if err = common.CopyFile(src, dst); err != nil {
+			return fmt.Errorf("error copying file: %w", err)
+		}
+	}
+
+	tar := archiver.NewTarGz()
+	tar.OverwriteExisting = true //
+	tar.MkdirAll = true          // make destination directory if it doesn't exist
+
+	if err = tar.Archive([]string{dir}, destination); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// fileNameWithoutExtension returns a filename with its extension stripped.
+func fileNameWithoutExtension(filename string) string {
+	base := filepath.Base(filename)
+	firstDot := strings.Index(base, ".")
+	if firstDot > -1 {
+		return base[:firstDot]
+	}
+	return base
+}
```

Comment:
Do you want `/path/to/whatever.tar.gz` to return `whatever` or `whatever.tar`?

Reply:
`whatever`

Reply:
OK, then this implementation is good 👍